src/main/java/core/basesyntax/service/impl/ReportWriterCsv.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportWriter;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class ReportWriterCsv implements ReportWriter {
+    @Override
+    public void writeReport(String report, String pathname) {
+        if (report == null || report.isEmpty()) {
+            throw new IllegalArgumentException("Data can't be null or 0!");
+        }
+
+        File file = new File(pathname);
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file))) {
+            bufferedWriter.write(report);
+        } catch (IOException ex) {
```

Comment:
Please be consistent and use one naming for exceptions (in some cases you have 'Exception' in others you have 'ex' please decide. I suggest using 'ex' or 'e'